api/v1/models/Upload.js

```diff
@@ -0,0 +1,33 @@
+var Model = require('./Model');
+var Upload = Model.extend({
+	tableName: 'uploads',
+	idAttribute: 'id',
+	hasTimestamps: ['created', 'updated']
+});
+
+/**
+ * Finds all uploads belonging to a given user
+ * @param  {User} owner		the owner of the uploads
+ * @param  {String} bucket	(optional) the bucket in which to search
+ * @return {Promise<Collection<Upload>>} all of the uploads belonging to the owner
+ */
+Upload.findByOwner = function (owner, bucket) {
+	var queryParams = { where : {} };
+
+	queryParams.where.owner_id = owner.get('id');
+	if (bucket) { queryParams.where.bucket = bucket; }
+
+	return Upload.collection().query(queryParams).fetch();
```

Comment:
[Seems like collection().fetch() is unfavored over model.fetchAll()](https://github.com/tgriesser/bookshelf/issues/595). Though they did say ASAP, but that was 1 year ago. Definitely something to think about.